sql/src/main/java/io/crate/expression/scalar/string/LpadRpadFunctionImplHelper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.data.Input;
+
+public class LpadRpadFunctionImplHelper {
```

Comment:
Not sure about the naming here.

I think something like `PadFunctions` or `StringPadding` would be sufficient. (And avoid the generic `Helper`).

Furthermore, `string1` and `string2` could be renamed into more concrete variables. E.g. `toPad` and `fillerChars` or something like that to give them some meaning.

For classes it's recommended to mark attributes if possible as `final` so that they can't be mutated. Other variables like `m` seem to be used exclusively in a local scope, so it's preferable to declare them within a local scope as well.

I'm generally wondering if using a class here is appropriate. Unless some JIT optimizations kick in to avoid heap allocations creating a new class instance to do the work of what could be one or two static methods seems a bit overkill (and is likely slower)

I'd also de-couple the `Input[] args` unwrapping from the actual padding implementation. So that there is like a `padStart(String toPad, int length, String fillerChars)` method. (That could be unit tested independently of our Scalar function framework around it).

If you look at the Guava `padStart` implementation the core logic is basically a

```
String.valueOf(c).repeat(Math.max(0, minimumLength - s.length())) + s;
```

I wonder if we could get closer to something like that. It will take a bit more logic due to supporting multiple filler chars instead of a single one, but we could probably still get away with a single loop?

Reply:
I de-coupled `Input[] args` unwrapping from actual implementation.

Reply:
Renamed and marked `final`.

`m`s value depends on `s1len` which `copyString` mutates.

Reply:
I used a helper class here to avoid introducing inheritance.

Reply:
The padding logic is port of [oracle_compat.c](https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/oracle_compat.c#L202) which wraps around at end of `fillChars`

Reply:
@mfussenegger thank you for having taken your time to review. I could get away with a single loop and get rid of helper class in favor of having return logic in both `lpad/rpad` implementations

```
Collections.nCopies(toPadLen / fillerLen, fillerLen)
    .forEach(n -> ret.append(fillerChars));
```